pallets/xcm-transactor/src/lib.rs

```diff
@@ -490,10 +518,128 @@ pub mod pallet {
+
+			let fee_location: MultiLocation =
+				T::CurrencyIdToMultiLocation::convert(fee_currency_id)
+					.ok_or(Error::<T>::NotCrossChainTransferableCurrency)?;
+
+			// Grab the destination
+			Self::transact_in_dest_chain_asset_signed(
+				dest.clone(),
+				who.clone(),
+				fee_location,
+				dest_weight,
+				call.clone(),
+				OriginKind::SovereignAccount,
+			)?;
+
+			// Deposit event
+			Self::deposit_event(Event::<T>::TransactedSigned {
+				fee_payer: who,
+				dest,
+				call,
+			});
+
+			Ok(())
+		}
+
+		/// Transact the call through the a signed origin in this chain
+		/// that should be converted to a transaction dispatch account in the destination chain
+		/// by any method implemented in the destination chains runtime
+		///
+		/// This time we are giving the currency as a multilocation instead of currencyId
+		#[pallet::weight(T::WeightInfo::transact_through_signed_multilocation())]
+		pub fn transact_through_signed_multilocation(
+			origin: OriginFor<T>,
+			dest: Box<VersionedMultiLocation>,
+			fee_location: Box<VersionedMultiLocation>,
+			dest_weight: Weight,
+			call: Vec<u8>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let dest = MultiLocation::try_from(*dest).map_err(|()| Error::<T>::BadVersion)?;
+			let fee_location =
+				MultiLocation::try_from(*fee_location).map_err(|()| Error::<T>::BadVersion)?;
+
+			// Grab the destination
+			Self::transact_in_dest_chain_asset_signed(
+				dest.clone(),
+				who.clone(),
+				fee_location,
+				dest_weight,
+				call.clone(),
+				OriginKind::SovereignAccount,
+			)?;
+
+			// Deposit event
+			Self::deposit_event(Event::<T>::TransactedSigned {
+				fee_payer: who,
+				dest,
+				call,
+			});
+
+			Ok(())
+		}
+
+		/// Set the fee per second of an asset on its reserve chain
+		#[pallet::weight(T::WeightInfo::set_fee_per_second())]
+		pub fn set_fee_per_second(
+			origin: OriginFor<T>,
+			asset_location: Box<VersionedMultiLocation>,
+			fee_per_second: u128,
+		) -> DispatchResult {
+			T::DerivativeAddressRegistrationOrigin::ensure_origin(origin)?;
+			let asset_location =
+				MultiLocation::try_from(*asset_location).map_err(|()| Error::<T>::BadVersion)?;
+
+			DestinationAssetFeePerSecond::<T>::insert(&asset_location, &fee_per_second);
+
+			Self::deposit_event(Event::DestFeePerSecondChanged {
+				location: asset_location,
+				fee_per_second,
+			});
+			Ok(())
+		}
+
+		/// Remove the fee per second of an asset on its reserve chain
+		#[pallet::weight(T::WeightInfo::set_fee_per_second())]
+		pub fn remove_fee_per_second(
+			origin: OriginFor<T>,
+			asset_location: Box<VersionedMultiLocation>,
+		) -> DispatchResult {
+			T::DerivativeAddressRegistrationOrigin::ensure_origin(origin)?;
+			let asset_location =
+				MultiLocation::try_from(*asset_location).map_err(|()| Error::<T>::BadVersion)?;
+
+			DestinationAssetFeePerSecond::<T>::remove(&asset_location);
```

Comment:
This can `Err` if not present. I think it's fine to silently ignore that case, but it might be good to mention the behavior in the fn's documentation.